test/e2e/common.go

```diff
@@ -251,28 +247,27 @@ func (s *linuxInstallerTestSuite) uninstall() {
 
 func (s *linuxInstallerTestSuite) assertUninstall() {
 	t := s.T()
-	t.Helper()
 	vm := s.Env().RemoteHost
 	t.Logf("Assert %s is removed", flavor)
 	// dd-agent user and config file should still be here
-	_, err := vm.Execute("id dd-agent")
-	assert.NoError(t, err, "user datadog-agent not present after remove")
-	assertFileExists(t, vm, fmt.Sprintf("/etc/%s/%s", s.baseName, s.configFile))
-	if flavor == "datadog-agent" {
-		// The custom file should still be here. All other files, including the extra integration, should be removed
-		assertFileExists(t, vm, fmt.Sprintf("%s/site-packages/testfile", s.getLatestEmbeddedPythonPath("datadog-agent")))
-		files := strings.Split(strings.TrimSuffix(vm.MustExecute("find /opt/datadog-agent -type f"), "\n"), "\n")
-		assert.Len(t, files, 1, fmt.Sprintf("/opt/datadog-agent present after remove, found %v", files))
-	} else {
-		// All files in /opt/datadog-agent should be removed
-		assertFileNotExists(t, vm, fmt.Sprintf("/opt/%s", s.baseName))
-	}
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		_, err := vm.Execute("id dd-agent")
+		assert.NoError(c, err, "user datadog-agent not present after remove")
+		assertFileExists(c, vm, fmt.Sprintf("/etc/%s/%s", s.baseName, s.configFile))
+		if flavor == "datadog-agent" {
+			// The custom file should still be here. All other files, including the extra integration, should be removed
+			assertFileExists(c, vm, fmt.Sprintf("%s/site-packages/testfile", s.getLatestEmbeddedPythonPath("datadog-agent")))
+			files := strings.Split(strings.TrimSuffix(vm.MustExecute("find /opt/datadog-agent -type f"), "\n"), "\n")
+			assert.Len(c, files, 1, fmt.Sprintf("/opt/datadog-agent present after remove, found %v", files))
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
			assert.Len(c, files, 1, fmt.Sprintf("/opt/datadog-agent present after remove , found %v, expected only %s, files, fmt.Sprintf("%s/site-packages/testfile", s.getLatestEmbeddedPythonPath("datadog-agent"))"))
```
You might store `fmt.Sprintf("%s/site-packages/testfile", s.getLatestEmbeddedPythonPath("datadog-agent"))"` into a variable as it is used above too

Reply:
Thank you ! I applied it